_books/ion-1-1/src/macros/system_macros.md

```diff
@@ -0,0 +1,398 @@
+## System Macros
+
+Many of the system macros _MAY_ be defined as template macros, and when possible, the specification includes a template.
+Templates are given here as normative example, but system macros are not required to be implemented as template macros.
+
+The macros that can be defined as templates are included as system macros because of their broad applicability, and
+so that Ion implementations can provide optimizations for these macros that can short-circuit the regular macro evaluation.
+
+### Stream Constructors
+
+#### `none`
+
+```ion
+(macro none () (.values))
+```
+
+`none` accepts no values and produces nothing (an empty stream).
+
+#### `values`
+
+```ion
+(macro values (v*) v)
+```
+
+This is, essentially, the identity function. 
+It produces a stream from any number of arguments, concatenating the streams produced by the nested expressions.
+Used to aggregate multiple values or sub-streams to pass to a single argument, or to return multiple results.
+
+#### `flatten`
+
+```ion
+(macro flatten (sequence+) /* Not representable in TDL */)
+```
+The `flatten` system macro constructs a stream from the content of one or more sequences.
+
+Produces a stream with the contents of all the `sequence` values.
+Any annotations on the `sequence` values are discarded.
+Any non-sequence arguments will raise an error.
+Any null arguments will be ignored.
+
+Examples:
+```ion
+(:flatten [a, b, c] (d e f)) => a b c d e f
+(:flatten [[], null.list] foo::()) => [] null.list
+```
+
+The `flatten` macro can also be used to splice the content of one list or s-expression into another list or s-expression.
+```ion
+[1, 2, (:flatten [a, b]), 3, 4] => [1, 2, a, b, 3, 4]
+```
+
+#### `parse_ion`
+
+Ion documents may be embedded in other Ion documents using the `parse_ion` macro.
+
+```ion
+(macro parse_ion (data!) /* Not representable in TDL */)
+```
+
+The `parse_ion` macro constructs a stream of values by parsing a blob literal or string literal as a single, self-contained Ion document.
+
+Embedded text example:
+```ion
+(:parse_ion
+    '''
+    $ion_1_1
+    $ion_encoding::((symbol_table ["foo" "bar"]]))
+    $1 $2
+    '''
+)
+=> foo bar
+```
+
+
+Embedded binary example:
+```ion
+(:parse_ion {{ 4AEB6qNmb2+jYmFy }} )
+=> foo bar
+```
+
+> [!IMPORTANT]
+> Unlike most macros, this macro specifically requires _literals_. Macros are not allowed to contain recursive calls,
+> and composing an embedded document from multiple expressions would make it possible to implement recursion in the
+> macro system.
+> 
+> Allowing context to leak from the outer scope into the document being parsed would also enable recursion.
+>
+> Furthermore, `parse_ion` cannot be invoked from a macro body (i.e. in Template Definition Language (TDL)). 
+> (Why not?)
+
+
+### Value Constructors
+
+#### `annotate`
+
+```ion
+(macro annotate (ann* value) /* Not representable in TDL */)
```

Comment:
If we decide to add footnotes, it might be helpful to explain that the annotations sequence comes first because it parallels how annotations are read from the data stream.